datastore/longhorn.go

```diff
@@ -6449,6 +6449,32 @@ func (s *DataStore) IsV2DataEngineDisabledForNode(nodeName string) (bool, error)
 	return false, nil
 }
 
+func (s *DataStore) IsVolumeCompatibleWithNodeEngine(volumeName string, nodeName string) (bool, error) {
+	// v1 volumes are always considered compatible.
+	// v2 volumes are compatible only if the node has NOT disabled v2 engine.
+
+	volume, err := s.GetVolumeRO(volumeName)
+	if err != nil {
+		return false, err
+	}
+
+	// If it's a v1 volume, v1 data engine is enabled in cluster
+	// no option for disable v1 engine on the specific node
+	// always compatible
+	if !types.IsDataEngineV2(volume.Spec.DataEngine) {
+		return true, nil
+	}
```

Comment:
NIT: I prefer to explictly check if the data engine is v1: `if types.IsDataEngineV1(volume.Spec.DataEngine) {`